src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,721 @@
+---
+title: Order view page new hooks demo tutorial 
+weight: 10
+---
+
+# Order view page new hooks demo tutorial
+{{< minver v="1.7.7.0" title="true" >}}
+
+## Introduction
+
+In this tutorial we are going to build a module to extend Order view page. 
+The module will add the following components:
+
+ - User Signature card below the Customer card.
+ 
+While creating this component you will learn how to:
+
+ - Use Doctrine (https://devdocs.prestashop.com/1.7/modules/concepts/doctrine/)
+ - Use Repository classes extending Symfony EntityRepository (https://symfony.com/doc/3.4/doctrine/repository.html)
+ - Use Symfony services (https://devdocs.prestashop.com/1.7/modules/concepts/services/)
+ - Use Twig templates to render HTML (https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/templating-with-twig/)
+ - Various design patterns: Repository, Factory, Presenter
+
+## Prerequisites
+
+- To be familiar with basic module creation.
+- To be familiar how Composer autoloads classes (https://devdocs.prestashop.com/1.7/modules/concepts/composer/)
+
+### Register hooks
+
+On module installation the following hooks can be registered:
+
+ - `displayBackOfficeOrderActions` - displayed between Customer and Messages cards in the Order page
+ - `displayAdminOrderTabContent` - for adding tab content to Order page
+ - `displayAdminOrderTabLink` - for adding tab links for tab content
+ - `displayAdminOrderMain` - for adding Order main information
+ - `displayAdminOrderSide` - for adding Order side information
+ - `displayAdminOrder` - displayed at the bottom of the Order page
+ - `displayAdminOrderTop` - displayed at the top of the Order page
+ 
+These hooks are visualized in the picture below:
+
+ {{< figure src="../img/view-order-hooks-demo/ps-view-order-page-hooks.jpg" title="Order page hooks locations" >}}
+ 
+ ```php
+ 
+    private function registerHooks(Module $module): bool
+    {
+        // All hooks in the order view page.
+        $hooks = [
+            'displayBackOfficeOrderActions',
+            'displayAdminOrderTabContent',
+            'displayAdminOrderTabLink',
+            'displayAdminOrderMain',
+            'displayAdminOrderSide',
+            'displayAdminOrder',
+            'displayAdminOrderTop',
+            'actionGetAdminOrderButtons',
+        ];
+
+        return (bool) $module->registerHook($hooks);
+    }
+ ```
+
+Let's start from the first one - `displayBackOfficeOrderActions` and create a demo for it. 
+
+### Create User Signature card below the Customer card.
+
+Lets create `Installer` class inside `/demovieworderhooks/src/Install` forlder structure
+ responsible for creating `signature` table in the database:
```

Comment:
Please explain why it's better to have this class outside of main module file (most PrestaShop modules put everything in the main module file). Basically a statement about SOLID principles, keeping classes and responsibilities separated 😄 

The idea is also to teach module developers our best practices